Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSFlippingRatioCorr.py

```diff
@@ -0,0 +1,347 @@
+                             'slit_i_upper_blade_position']
+    input_workspaces = {}
+    sf_outws_name = None
+    nsf_outws_name = None
+    dfr = 0.05
+
+    def category(self):
+        """
+        Returns category
+        """
+        return 'PythonAlgorithms\\MLZ\\DNS\\CorrectionFunctions'
+
+    def name(self):
+        """
+        Returns name
+        """
+        return "DNSFlippingRatioCorr"
+
+    def summary(self):
+        return "Peforms flipping ratio correction on a given dataset using the NiCr data."
+
+    def PyInit(self):
+        self.declareProperty(MatrixWorkspaceProperty("SFDataWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with spin-flip experimental data from sample.")
+        self.declareProperty(MatrixWorkspaceProperty("NSFDataWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with non spin-flip experimental data from sample.")
+        self.declareProperty(MatrixWorkspaceProperty("SFNiCrWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with spin-flip NiCr data.")
+        self.declareProperty(MatrixWorkspaceProperty("NSFNiCrWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with non spin-flip NiCr data.")
+        self.declareProperty(MatrixWorkspaceProperty("SFBkgrWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with spin-flip background for NiCr.")
+        self.declareProperty(MatrixWorkspaceProperty("NSFBkgrWorkspace", "", direction=Direction.Input),
+                             doc="A workspace with non spin-flip background for NiCr.")
+        self.declareProperty(MatrixWorkspaceProperty("SFOutputWorkspace", "", direction=Direction.Output),
+                             doc="A workspace to save the corrected spin-flip data.")
+        self.declareProperty(MatrixWorkspaceProperty("NSFOutputWorkspace", "", direction=Direction.Output),
+                             doc="A workspace to save the corrected non spin-flip data.")
+        self.declareProperty("DoubleSpinFlipScatteringProbability", 0.05, FloatBoundedValidator(lower=0, upper=1.0),
+                             doc="Probability of the double spin-flip scattering. Number between 0 and 1.")
+
+        return
+
+    def _dimensions_valid(self):
+        """
+        Checks validity of the workspace dimensions:
+        all given workspaces must have the same number of dimensions
+        and the same number of histograms
+        and the same number of bins
+        """
+        ndims = []
+        nhists = []
+        nbins = []
+        for wsname in self.input_workspaces.values():
+            ws = api.AnalysisDataService.retrieve(wsname)
+            ndims.append(ws.getNumDims())
+            nhists.append(ws.getNumberHistograms())
+            nbins.append(ws.blocksize())
+
+        nd = ndims[0]
+        nh = nhists[0]
+        nb = nbins[0]
```

Comment:
These 2-character variable names are most likely what pylint didn't like ('invalid-name' disable) (also the ws variable, line 71). I don't think this is really important, so just ignore or fix it as you prefer. 

All the code is well organized and the documentation is also very good, so this looks ready to go when you fix the merge conflict.